### PR TITLE
[TP-179336] Use postmessage instead of storage event for embedded case

### PR DESCRIPTION
**Pull request by Gaurav0**

### CODE REVIEW

File: addon/mixins/ui-service-mixin.js [hunk 1]
```diff
@@ -55,7 +56,36 @@ var ServicesMixin = Mixin.create({
           });
         }
       };
+
+      // Using postMessage as an alternative to localStorage/storageEvent
```
Comment (AndrewHYi) [inline addon/mixins/ui-service-mixin.js:60]: Let's add some updated tests  - QA is getting pretty crazy https://civicplus.tpondemand.com/entity/179336-open-new-window-vs-modal-popup/comments/309113.

**Review by AndrewHYi — CHANGES_REQUESTED** [review]

**Review by AndrewHYi — CHANGES_REQUESTED** [review]
suggestion: Is there a way we can run/update the torii tests?

**Comment by zbeat** [comment]
TP story link:
https://civicplus.tpondemand.com/entity/179336-open-new-window-vs-iframe-modal

We need to fix the TP<>Github integration permissions, so all SCF repos are included.

**Comment by Gaurav0** [comment]
@AndrewHYi Tests are running automatically.

**Comment by AndrewHYi** [comment]
@Gaurav0 nice, looks like existing tests pass - should we add tests for the new scenarios?

File: addon/mixins/ui-service-mixin.js (hunk 1 above)
Comment (AndrewHYi) [reply addon/mixins/ui-service-mixin.js:60]: Nice, tests!

File: addon/mixins/ui-service-mixin.js [hunk 2]
```diff
@@ -60,25 +60,23 @@ var ServicesMixin = Mixin.create({
...
-              decodeURIComponent(key)
-            );
-            if (remoteId === remoteIdFromEvent) {
-              var data = parseMessage(msg[key], keys);
-              localStorage.removeItem(key);
-              run(function () {
-                resolve(data);
-              });
-            }
+        if (messageEvent.data === 'getPendingRequestKey') {
```
Comment (AndrewHYi) [inline addon/mixins/ui-service-mixin.js:63]: suggestion: this fix should be merged alongside the first commit. Also, the commit about feature flags is confusing here, but that only makes sense in the context of the SCF application (not here in Torii fork land).

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "torii",
-  "version": "1.0.0-beta.2",
+  "version": "1.0.0-beta.3",
```
Comment (AndrewHYi) [inline package.json:3]: suggestion: the version bump should be the last commit - also, do we need to bump the version up again?

File: tests/unit/mixins/ui-service-mixin-test.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
+/* eslint-disable ember/no-mixins, ember/no-new-mixins, ember/no-classic-classes */
+
+import EmberObject from '@ember/object';
```
Comment (AndrewHYi) [inline tests/unit/mixins/ui-service-mixin-test.js:3]: suggestion: let's add the test to the first commit.

thought: I need some help walking through the test code - I'll schedule something for this afternoon.

Update: tests seem to be failing - can reschedule the group review we rescheduled if you need more time.

**Review by AndrewHYi — CHANGES_REQUESTED** [review]

**Review by AndrewHYi — APPROVED** [review]

**Pull request merged**